src/sentry/middleware/access_log.py

```diff
@@ -0,0 +1,85 @@
+from __future__ import annotations
+
+import logging
+import time
+from dataclasses import dataclass
+
+from django.utils.deprecation import MiddlewareMixin
+from rest_framework.request import Request
+from rest_framework.response import Response
+
+api_access_logger = logging.getLogger("sentry.access.api")
+
+
+@dataclass
+class _AccessLogMetaData:
+    request_start_time: float | None
+
+    def get_request_duration(self) -> float | None:
+        return time.time() - self.request_start_time if self.request_start_time else None
+
+
+_empty_request_metadata = _AccessLogMetaData(request_start_time=None)
+
+
+class AccessLogMiddlewware(MiddlewareMixin):
+    def _get_token_name(self, request: Request):
+        auth = getattr(request, "auth", None)
+        if not auth:
+            return None
+        token_class = getattr(auth, "__class__", None)
+        return token_class.__name__ if token_class else None
```

Comment:
Ok so after more investigation, I don't know if this information is ever there. Do you know where it gets attached @AniketDas-Tekky ?

Reply:
Looks like it's in the Authentication Middleware. So this will need to run after authentication